### PR TITLE
Fix aws spot price with default config

### PR DESCRIPTION
**Pull request by sachin-rafay**
## What does this PR change?
* Currently, when user does not provide spot data feed but uses spot instances, the code picks hardcoded spot cpu cost. The solution here is to pick the public pricing for spot instances with default config (no spot data feed configured).

## Does this PR relate to any other PRs?
* I hope not.

## How will this PR impact users?
* When user does not provide spot data feed but uses opencost on spot instances, public pricing will be picked.

## Does this PR address any GitHub or Zendesk issues?
* Closes ... https://github.com/opencost/opencost/issues/2243

## How was this PR tested?
* The changes were tested on AWS spot instances

## Does this PR require changes to documentation?
* I hope not. When the user does not configure spot data feed, opencost should be expected to provide public pricing.

## Have you labeled this PR and its corresponding Issue as "next release" if it should be part of the next OpenCost release? If not, why not?
* Release eligibility has not yet been determined.

### CODE REVIEW

File: pkg/cloud/aws/provider.go
Comment (thomasvn) [inline pkg/cloud/aws/provider.go]: In the case that a Spot Data Feed isn't available, I'm unsure if we should be creating a dependency on the AWS OnDemand Pricing Sheet?

Comment (thomasvn) [reply pkg/cloud/aws/provider.go]: Or maybe in the case that `!publicPricingFound`, we can set use defaults here instead of throwing an error?

Comment (sachin-rafay) [reply pkg/cloud/aws/provider.go]: Thanks for the review @thomasvn ! I have made the change to return defaults when public price is not available.

I have made one change while returning defaults though: I am returning both vCPU and RAM default price instead of just vCPU price like earlier. If you want me to change that, please let me know. Thanks!

**Comment by mattray** [comment]
Thanks for the PR @sachin-rafay! @opencost/opencost-maintainers could we get a review to followup with @thomasvn's review? This seems pretty fundamental

File: pkg/cloud/aws/provider.go
Comment (thomasvn) [inline pkg/cloud/aws/provider.go]:
```suggestion
			log.DedupedWarningf(5, "Could not find Node %s's public pricing info, using default configured spot prices instead", k.ID())
```

Thoughts?

File: pkg/cloud/aws/provider.go
Comment (thomasvn) [reply pkg/cloud/aws/provider.go]: Thanks for this change. I'm in agreement here!

**Comment by thomasvn** [comment]
These changes LGTM. We have some nightly integration tests which validate pod allocation costs scheduled on an OnDemand EKS node. I'll make sure that remains unchanged after merge of this PR.

**Review by thomasvn — APPROVED** [review]

**Comment by JonCholas** [comment]
Any plan to merge this soon? this is blocking us to adopt kubecost for all our clusters

**Review by AjayTripathy — APPROVED** [review]

**Comment by AjayTripathy** [comment]
Thank you so much for the review! @thomasvn and the PR @sachin-rafay . Merging now. Expect this in our release of 2.0 scheduled for late January.

**Pull request merged**
